integration/test_collection_model.py

```diff
@@ -252,3 +255,61 @@ class TestTenantSearch(BaseProperty):
 
     objects2 = tenant2.query.bm25_flat(query="some", return_metadata=MetadataQuery(uuid=True))
     assert len(objects2) == 0
+
+
+def make_list() -> List[int]:
+    return []
+
+
+@pytest.mark.parametrize(
+    "member_type, value_to_add,default,default_factory, exception",
+    [
+        (int, 10, 5, None, True),
+        (int, 10, PydanticUndefined, None, True),
+        (Optional[int], 10, PydanticUndefined, None, False),
+        (Optional[int], 10, None, None, False),
+        (Optional[int], 10, 10, None, False),
+        (List[int], [10], None, None, True),
+        (Optional[List[int]], [10], None, None, False),
+        (List[int], [10], None, make_list, True),
+    ],
+)
+def test_update_properties(
+    client: weaviate.Client,
+    member_type: type,
+    value_to_add,
+    default,
+    default_factory,
+    exception: bool,
+):
+    def create_original_collection():
+        # class definition will be gone when this is out of scope
+        class TestPropUpdate(BaseProperty):
+            name: str
+
+        client.collection_model.delete(TestPropUpdate)
+        collection_first = client.collection_model.create(
+            CollectionModelConfig(model=TestPropUpdate, vectorizer=Vectorizer.NONE)
+        )
+        collection_first.data.insert(TestPropUpdate(name="first"))
+
+    create_original_collection()
+
+    field = Field()
+    if default_factory is not None:
+        field = Field(default_factory=default_factory)
+    elif default is not None:
+        field = Field(default=default)
+
+    class TestPropUpdate(BaseProperty):
+        name: str
+        number: member_type = field
+
+    if exception:
+        with pytest.raises(WeaviateAddProperty):
+            client.collection_model.update_property(TestPropUpdate)
+    else:
+        collection = client.collection_model.update_property(TestPropUpdate)
+        collection.data.insert(TestPropUpdate(name="second", number=value_to_add))
+        objects = collection.data.get()
+        assert len(objects) == 2
```

Comment:
Should we also assert the attributes and values of these two collection objects?

Reply:
added